bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/Storage.java

```diff
@@ -1115,7 +1115,7 @@ File stageContent(final InputStream in, final URL sourceURL) throws BundleExcept
 
 	File stageContent0(InputStream in, URL sourceURL) throws BundleException {
 		File outFile = null;
-		try {
+		try (InputStream stream = in) {
```

Comment:
Before 68fa0c03747cf77307616b0fc644757a81e39578 the given `InputStream` was closed, in case the `sourceURL` uses the file protocol and the specified file (`inFile`) is not a directory.
It wasn't closed if `inFile` was a directory (was this a bug?).
After the mentioned commit the `InputStream` was only closed if another protocol than `file` was used.
With this change it is always closed-